### ISSUE TITLE
Gatekeeper: “nvALT.app” can’t be opened because the identity of the developer cannot be confirmed.

### ISSUE THREAD

**Issue by frankrausch** [ttscoff/nv#423]
I get this message for v2.2.7 (125) on two different computers after:
- Downloaded `.dmg` from http://brettterpstra.com/2016/11/14/nvalt-2-dot-2-7-everythingll-be-fine/
- Dragged to `/Applications` folder
- Launched nvALT

<img width="532" alt="screen shot 2016-12-04 at 13 12 46" src="https://cloud.githubusercontent.com/assets/6783714/20866070/69512eba-ba23-11e6-8b99-e5df1e005303.png">

**Comment by sdonovan1985** [comment]
This is normal, if you haven't adjusted security settings. Go to the "Security and Privacy" panel in System Preferences, in the general tab, set "Allow apps downloaded from:" to Anywhere.

**Comment by frankrausch** [comment]
Thanks, but I’m afraid this is not great advice for most users. :)
The system setting should never be changed to allow launching unsigned apps globally.

But yeah, since code signing is broken for nvALT at the moment, right-clicking the .app and choosing “Open” does the trick (and circumvents Gatekeeper for just this one app).

**Comment by sdonovan1985** [comment]
Ah, that's what you were after! I follow now.

**Comment by frankrausch** [comment]
You’re right, I should change the title to reflect that this issue is about Gatekeeper being broken (and that I’m not looking for a workaround).

**Comment by ttscoff** [comment]
Yeah, I'm still wracking my brain to figure out why this happens. The 
app is codesigned with a wildcard developer ID that I use successfully 
with other distributions, and the binary passes `codesign` validation. I 
think one of the bundled binaries may be causing this.

-Brett

**Comment by ksuther** [comment]
@ttscoff Here's the likely issue:

`spctl --verbose=4 --assess --type execute nvALT.app`:
`/Applications/nvALT.app/: rejected (invalid destination for symbolic link in bundle)`

Specifically, there's broken symlinks in Sparkle:

```
Sparkle.framework//Versions/A/Resources:
lrwxr-xr-x  1 kent  admin     84 Nov  1 12:44 fr_CA.lproj -> /Users/andym/Development/Build Products/Release/Sparkle.framework/Resources/fr.lproj

Sparkle.framework//Versions/A/Resources/fr.lproj:
lrwxr-xr-x  1 kent  admin    84 Nov  1 12:44 fr.lproj -> /Users/andym/Development/Build Products/Release/Sparkle.framework/Resources/fr.lproj
```

**Comment by ttscoff** [comment]
Ok, I found those symbolic links and removed, checked the built product 
for any other broken links, and ran spctl again, still rejected (but 
without any helpful error message). Any other thoughts?

-Brett

**Comment by ksuther** [comment]
You can give codesign a shot and see if it comes up with anything:

`codesign -vvvvv --deep --strict /Applications/nvALT.app`

With the broken symlinks I couldn't get it to run cleanly (presumably it was choking on them), but maybe that'll turn something up.

**Comment by ttscoff** [comment]
Woohoo, I got it! I'll put an update out shortly.

-=
